llarp/service/Info.hpp

```diff
@@ -5,7 +5,11 @@
 #include <service/types.hpp>
 #include <util/bencode.hpp>
 
+#if __cplusplus >= 201703L
+#include <optional>
```

Comment:
The entire point of doing this was to have a consistent api across all platforms, so the lokinet codebase is "clean" and we push all that jank into the 3rd party libraries where possible.

Reply:
>macintosh only
```cpp
// ABSL_HAVE_STD_OPTIONAL
//
// Checks whether C++17 std::optional is available.
#ifdef ABSL_HAVE_STD_OPTIONAL
#error "ABSL_HAVE_STD_OPTIONAL cannot be directly set."
#endif

#ifdef __has_include
#if __has_include(<optional>) && __cplusplus >= 201703L && \
    ABSL_INTERNAL_MACOS_HAS_CXX_17_TYPES
#define ABSL_HAVE_STD_OPTIONAL 1
#endif
#endif
```
No.

Reply:
That gets set later in:

https://github.com/abseil/abseil-cpp/blob/d78310fe5a82f2e0e6e16509ef8079c8d7e4674e/absl/base/config.h#L435-L448

Reply:
this project does not support Microsoft C++...had to abandon it early on because `getopt(3)` 

```c
#if defined(_MSC_VER) && _MSC_VER >= 1910 && ((defined(_MSVC_LANG) && _MSVC_LANG > 201402) || __cplusplus > 201402)
```